src/importlinter/contracts/layers.py

```diff
@@ -210,8 +207,8 @@ def _get_undeclared_modules(self, graph: grimp.ImportGraph) -> set[str]:
         undeclared_modules = set()
 
         exhaustive_ignores: set[str] = self.exhaustive_ignores or set()  # type: ignore
-        layers: set[str] = {layer.name for layer in self.flattened_layers}
-        declared_modules = layers | exhaustive_ignores
+        module_tails: set[str] = {module_tail.name for module_tail in self.flattened_module_tails}
+        declared_modules = module_tails | exhaustive_ignores
```

Comment:
Could we rename `declared_modules` to  `declared_module_tails` (now we have a name for it)? And, lower down, could we rename `undotted_module` to `module_tail`?